specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2019-08-01/managedClusters.json

```diff
@@ -0,0 +1,2349 @@
+        },
+        "dnsPrefix": {
+          "type": "string",
+          "description": "DNS prefix specified when creating the managed cluster."
+        },
+        "fqdn": {
+          "readOnly": true,
+          "type": "string",
+          "description": "FQDN for the master pool."
+        },
+        "agentPoolProfiles": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedClusterAgentPoolProfile"
+          },
+          "description": "Properties of the agent pool."
+        },
+        "linuxProfile": {
+          "$ref": "#/definitions/ContainerServiceLinuxProfile",
+          "description": "Profile for Linux VMs in the container service cluster."
+        },
+        "windowsProfile": {
+          "$ref": "#/definitions/ManagedClusterWindowsProfile",
+          "description": "Profile for Windows VMs in the container service cluster."
+        },
+        "servicePrincipalProfile": {
+          "$ref": "#/definitions/ManagedClusterServicePrincipalProfile",
+          "description": "Information about a service principal identity for the cluster to use for manipulating Azure APIs."
+        },
+        "addonProfiles": {
+          "additionalProperties": {
+            "$ref": "#/definitions/ManagedClusterAddonProfile"
+          },
+          "description": "Profile of managed cluster add-on."
+        },
+        "nodeResourceGroup": {
+          "type": "string",
+          "description": "Name of the resource group containing agent pool nodes."
+        },
+        "enableRBAC": {
+          "type": "boolean",
+          "description": "Whether to enable Kubernetes Role-Based Access Control."
+        },
+        "enablePodSecurityPolicy": {
+          "type": "boolean",
+          "description": "(PREVIEW) Whether to enable Kubernetes Pod security policy."
+        },
+        "networkProfile": {
+          "$ref": "#/definitions/ContainerServiceNetworkProfile",
+          "description": "Profile of network configuration."
+        },
+        "aadProfile": {
+          "$ref": "#/definitions/ManagedClusterAADProfile",
+          "description": "Profile of Azure Active Directory configuration."
+        },
+        "privateLinkResources": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ManagedClusterPrivateLinkResource"
+          },
+          "description": "Properties related to private link cluster."
+        },
+        "apiServerAccessProfile": {
+          "$ref": "#/definitions/ManagedClusterAPIServerAccessProfile",
+          "description": "Access profile for managed cluster API server."
+        }
+      },
+      "description": "Properties of the managed cluster."
+    },
+    "ManagedClusterPrivateLinkResource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "The fully qualified Azure resource ID."
+        },
+        "name": {
```

Comment:
name can be obtained from the id. why do you need user to pass in name, type etc. when all these can be obtained from the id.

Reply:
this is not ask for user input but allows AKS RP to pass back the info to customer(NRP in this particular case)

Reply:
are these read-only properties then? if so, can you please mark them as such

Reply:
@weinong  your input is needed here

Reply:
we decided to remove this in this version.